src/org/pentaho/reporting/platform/plugin/PluginPerfomanceMonitorContext.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License, version 2 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/gpl-2.0.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ *
+ * Copyright 2006 - 2016 Pentaho Corporation.  All rights reserved.
+ */
+
+package org.pentaho.reporting.platform.plugin;
+
+import org.pentaho.platform.api.engine.IPentahoSession;
+import org.pentaho.platform.engine.core.system.PentahoSessionHolder;
+import org.pentaho.reporting.engine.classic.core.states.PerformanceMonitorContext;
+import org.pentaho.reporting.libraries.base.util.LoggingStopWatch;
+import org.pentaho.reporting.libraries.base.util.PerformanceLoggingStopWatch;
+import org.pentaho.reporting.platform.plugin.async.ReportListenerThreadHolder;
+
+import javax.swing.event.ChangeListener;
+
+/**
+ * Created by dima.prokopenko@gmail.com on 4/19/2016.
+ */
+public class PluginPerfomanceMonitorContext implements PerformanceMonitorContext {
+
+  public static final String FORMAT = " sessionid[%1s] userid[%2s] requestid[%3s]";
+
+  @Override public PerformanceLoggingStopWatch createStopWatch( String tag ) {
+    return new LoggingStopWatch( tag, this.getMessage() );
+  }
+
+  @Override public PerformanceLoggingStopWatch createStopWatch( String tag, Object message ) {
+    return new LoggingStopWatch( tag, message + this.getMessage() );
```

Comment:
Instead of using string + string, pass the "message" object into the "getMessage" method and format everything in one operation. 

Your format string would then be "%s sessionid[%s] userid[%s] requestid[%s]"

You probably dont intend to use a one-space padding, two-space padding and 3 space padding on the format string either.